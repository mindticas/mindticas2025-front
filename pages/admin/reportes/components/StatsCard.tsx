import { Box, Flex, Text } from '@chakra-ui/react';
import { ComponentType, SVGProps } from 'react';

interface StatsCardProps {
    title: string;
    value: number;
    icon?: ComponentType<SVGProps<SVGSVGElement>>;
    color?: string;
}

const StatsCard = ({
    title,
    value,
    icon: IconComponent,
    color,
}: StatsCardProps) => {
    const bgColor = `${color?.split('.')[0]}.50`;
    return (
        <Box
            shadow='md'
            p='4'
            w='100%'
            alignContent='center'
            textAlign='center'
            borderRadius='lg'
            bg={bgColor}
            borderLeft='8px solid'
            borderLeftColor={color}
            transition='all 0.3s'
            _hover={{
                transform: 'translateY(-5px)',
                shadow: 'lg',
            }}
            backdropFilter='blur(8px)'
        >
            <Flex justifyContent='center' mb='3'>
                <Box bg={color} p='3' borderRadius='lg' width='fit-content'>
                    {IconComponent && <IconComponent color='white' />}
                </Box>
            </Flex>
            <Text fontSize='md' fontWeight='medium' color='gray.700' mb='1'>
                {title}
            </Text>
            <Text fontSize='2xl' fontWeight='bold'>
                {value}
            </Text>
        </Box>
    );
};

export default StatsCard;
