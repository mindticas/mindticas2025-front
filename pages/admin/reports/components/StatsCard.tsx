import { Box, Flex, Spinner, Text } from '@chakra-ui/react';
import { ComponentType, SVGProps } from 'react';

interface StatsCardProps {
    title: string;
    value: number;
    icon?: ComponentType<SVGProps<SVGSVGElement>>;
    color?: string;
    isLoading?: boolean;
}

const StatsCard = ({
    title,
    value,
    icon: IconComponent,
    color,
    isLoading = false,
}: StatsCardProps) => {
    const bgColor = `${color?.split('.')[0]}.50`;
    return (
        <Box
            shadow='md'
            p='4'
            w='100%'
            alignContent='center'
            textAlign='center'
            borderRadius='lg'
            bg={bgColor}
            borderLeft='8px solid'
            borderLeftColor={color}
            transition='all 0.3s'
            _hover={{
                transform: 'translateY(-5px)',
                shadow: 'lg',
            }}
            backdropFilter='blur(8px)'
        >
            <Flex justifyContent='center' mb='3'>
                <Box bg={color} p='3' borderRadius='lg' width='fit-content'>
                    {IconComponent && <IconComponent color='white' />}
                </Box>
            </Flex>
            <Text fontSize='md' fontWeight='medium' color='gray.700' mb='1'>
                {title}
            </Text>
            <Flex justifyContent='center' alignItems='center' height='36px'>
                {isLoading ? (
                    <Spinner size='lg' color={color} />
                ) : (
                    <Text fontSize='2xl' fontWeight='bold'>
                        {value}
                    </Text>
                )}
            </Flex>
        </Box>
    );
};

export default StatsCard;
