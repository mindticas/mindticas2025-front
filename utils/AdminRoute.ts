import Layout from '@/components/Layout';
import { Box } from '@chakra-ui/react';

export class AdminRoute {
    // Fixed route for admin
    private static route: string = '/admin';

    // Returns the Layout component if the pathname contains '/admin', otherwise returns box
    static getLayout(pathname: string) {
        return pathname.includes(this.route) ? Layout : Box;
    }
}
