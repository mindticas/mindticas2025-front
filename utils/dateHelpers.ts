// Gets the name of the day according to the full calendar format
export const getDayName = (dateStr: string): string => {
    const [year, month, day] = dateStr.split('-').map(Number)
    const date = new Date(year, month -1, day)
    return date.toLocaleDateString('es-ES', { weekday: 'long'})
}
// Get the milliseconds in a day
export function addDays(date: Date, days: number): Date {
    const milisecondsPerDay = 24 * 60 * 60 * 1000; 
    return new Date(date.getTime() + days * milisecondsPerDay)
}