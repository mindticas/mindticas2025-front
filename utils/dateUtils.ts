import { DateTime } from "luxon";

interface DateTimeParts {
    date: string;
    time: string;
}

// * @returns Object with (YYYY-MM-DD) y time (HH:MM)

export const splitDateTimeFromISO = (
    isoString: string | null | undefined,
): DateTimeParts => {
    if (!isoString) return { date: '', time: '' };

    const dateObj = new Date(isoString);

    if (isNaN(dateObj.getTime())) {
        return { date: '', time: '' };
    }

    // format date (YYYY-MM-DD)
    const date = DateTime.fromJSDate(dateObj)
       .setZone('local')
        .toFormat('yyyy-MM-dd');

    // format time (HH:MM)
    const time = dateObj
        .toLocaleTimeString('es-ES', {
            hour: '2-digit',
            minute: '2-digit',
            hour12: false,
        })
        .replace(/\./g, '');

    return { date, time };
};
