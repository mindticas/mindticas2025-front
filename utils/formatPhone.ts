export default function formatPhone(phone: string): string {
    // Remove everything that is not digits
    const digits = phone.replace(/\D/g, '');
    // Applies format only if it has 10 digits
    if (digits.length === 10) {
        return `${digits.slice(0, 3)}-${digits.slice(3, 6)}-${digits.slice(6)}`;
    }
    return phone; // Returns the original if it doesn't match
}
