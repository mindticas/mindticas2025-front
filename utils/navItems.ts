import { BarChart, Calendar, Scissors, UserCog, Users } from 'lucide-react';

export const navItems = [
    { name: 'Citas', href: '/admin/', icon: Calendar },
    { name: 'Clientes', href: '/admin/clients', icon: Users },
    { name: 'Empleados', href: '/admin/employees', icon: UserCog },
    { name: 'Tratamientos', href: '/admin/treatments', icon: Scissors },
    { name: 'Reportes', href: '/admin/reports', icon: BarChart },
];
