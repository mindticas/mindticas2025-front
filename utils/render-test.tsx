import { Provider } from '@/components/ui/provider';
import { BookingProvider } from '@/context/BookingContext';
import { render as rtlRender } from '@testing-library/react';

export function render(ui: React.ReactNode) {
    return rtlRender(<>{ui}</>, {
        wrapper: (props: React.PropsWithChildren) => (
            <Provider>
                <BookingProvider>{props.children}</BookingProvider>
            </Provider>
        ),
    });
}
