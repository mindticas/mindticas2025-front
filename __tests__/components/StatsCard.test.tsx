import { DollarSign } from 'lucide-react';
import { render } from '@/utils/render-test';
import StatsCard from '@/pages/admin/reports/components/StatsCard';

describe('StatsCard Snapshot', () => {
    it('renders correctly with icon and value', () => {
        const { asFragment } = render(
            <StatsCard
                title='Ganancias'
                value={1234}
                icon={DollarSign}
                color='green.500'
                isLoading={false}
            />,
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders loading state correctly', () => {
        const { asFragment } = render(
            <StatsCard
                title='Ganancias'
                value={0}
                icon={DollarSign}
                color='green.500'
                isLoading={true}
            />,
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders correctly without icon or color', () => {
        const { asFragment } = render(
            <StatsCard title='Servicios' value={42} isLoading={false} />,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
