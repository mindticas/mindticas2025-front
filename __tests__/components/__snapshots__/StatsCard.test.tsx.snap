// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatsCard Snapshot renders correctly with icon and value 1`] = `
<DocumentFragment>
  <script
    nonce=""
  >
    ((e,s,n,m,o,l,u,h)=&gt;{let a=document.documentElement,v=["light","dark"];function p(c){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&l?o.map(f=&gt;l[f]||f):o;k?(a.classList.remove(...S),a.classList.add(c)):a.setAttribute(y,c)}),C(c)}function C(c){h&&v.includes(c)&&(a.style.colorScheme=c)}function d(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(m)p(m);else try{let c=localStorage.getItem(s)||n,y=u&&c==="system"?d():c;p(y)}catch(c){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <div
    class="css-1b8npii"
  >
    <div
      class="css-1y96gi1"
    >
      <div
        class="css-1cteixr"
      >
        <svg
          class="lucide lucide-dollar-sign"
          fill="none"
          height="24"
          stroke="white"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="12"
            x2="12"
            y1="2"
            y2="22"
          />
          <path
            d="M17 5H9.5a3.5 3.5 0 0 0 0 7h5a3.5 3.5 0 0 1 0 7H6"
          />
        </svg>
      </div>
    </div>
    <p
      class="css-ycbeac"
    >
      Ganancias
    </p>
    <div
      class="css-1vhhqj7"
    >
      <p
        class="css-158617b"
      >
        1234
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`StatsCard Snapshot renders correctly without icon or color 1`] = `
<DocumentFragment>
  <script
    nonce=""
  >
    ((e,s,n,m,o,l,u,h)=&gt;{let a=document.documentElement,v=["light","dark"];function p(c){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&l?o.map(f=&gt;l[f]||f):o;k?(a.classList.remove(...S),a.classList.add(c)):a.setAttribute(y,c)}),C(c)}function C(c){h&&v.includes(c)&&(a.style.colorScheme=c)}function d(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(m)p(m);else try{let c=localStorage.getItem(s)||n,y=u&&c==="system"?d():c;p(y)}catch(c){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <div
    class="css-gsefhn"
  >
    <div
      class="css-1y96gi1"
    >
      <div
        class="css-axop6m"
      />
    </div>
    <p
      class="css-ycbeac"
    >
      Servicios
    </p>
    <div
      class="css-1vhhqj7"
    >
      <p
        class="css-158617b"
      >
        42
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`StatsCard Snapshot renders loading state correctly 1`] = `
<DocumentFragment>
  <script
    nonce=""
  >
    ((e,s,n,m,o,l,u,h)=&gt;{let a=document.documentElement,v=["light","dark"];function p(c){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&l?o.map(f=&gt;l[f]||f):o;k?(a.classList.remove(...S),a.classList.add(c)):a.setAttribute(y,c)}),C(c)}function C(c){h&&v.includes(c)&&(a.style.colorScheme=c)}function d(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(m)p(m);else try{let c=localStorage.getItem(s)||n,y=u&&c==="system"?d():c;p(y)}catch(c){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <div
    class="css-1b8npii"
  >
    <div
      class="css-1y96gi1"
    >
      <div
        class="css-1cteixr"
      >
        <svg
          class="lucide lucide-dollar-sign"
          fill="none"
          height="24"
          stroke="white"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="12"
            x2="12"
            y1="2"
            y2="22"
          />
          <path
            d="M17 5H9.5a3.5 3.5 0 0 0 0 7h5a3.5 3.5 0 0 1 0 7H6"
          />
        </svg>
      </div>
    </div>
    <p
      class="css-ycbeac"
    >
      Ganancias
    </p>
    <div
      class="css-1vhhqj7"
    >
      <span
        class="chakra-spinner css-15lzuyp"
      />
    </div>
  </div>
</DocumentFragment>
`;
